mathics/core/evaluators.py

```diff
@@ -24,20 +24,20 @@
 )
 
 
-def apply_N(expression, evaluation, prec=SymbolMachinePrecision):
+def eval_N(expression, evaluation, prec=SymbolMachinePrecision):
     """
     Equivalent to Expression("N", expression).evaluate(evaluation)
     """
     evaluated_expression = expression.evaluate(evaluation)
-    result = apply_nvalues(evaluated_expression, prec, evaluation)
+    result = apply_N(evaluated_expression, prec, evaluation)
     if result is None:
         return expression
     if isinstance(result, Number):
         return result
     return result.evaluate(evaluation)
 
 
-def apply_nvalues(expr, prec, evaluation):
+def apply_N(expr, prec, evaluation):
```

Comment:
Note that evaluation has the last two parameters swapped. What's up here?

Reply:
The order of the parameters in `apply_N` (former `apply_nvalues`)  is just the order that it had when it was the method `N.apply_with_prec`. On the other hand, the order in `eval_N` (former `apply_N`) is inverted because I wanted to put `precision` as an optional parameter, and then, it should go after  `evaluation`, which is not optional.

I also used that to check if the names at some places were confused (because if it does, it will raise errors).

Reply:
Thanks for the explanation. This however then points out that this irregular and confusing. 

For now, I think `apply_N`  should be called `apply_N_with_prec`. (And we should rename `N.apply_with_prec` to `N.apply_N_with_prec` as well. 

> I also used that to check if the names at some places were confused (because if it does, it will raise errors).

This might seem like a clever trick or hack, but it's really the wrong way to handle the problem. The right way is to add types to the arguments and use flake8 or pylint to catch type errors rather than making parameter ordering.

Reply:
> Thanks for the explanation. This however then points out that this irregular and confusing.
> 
> For now, I think `apply_N` should be called `apply_N_with_prec`. (And we should rename `N.apply_with_prec` to `N.apply_N_with_prec` as well.
> 

So, the idea would be to do not have `prec` as an optional argument, isn't it?


> > I also used that to check if the names at some places were confused (because if it does, it will raise errors).
> 
> This might seem like a clever trick or hack, but it's really the wrong way to handle the problem. The right way is to add types to the arguments and use flake8 or pylint to catch type errors rather than making parameter ordering.

Yes, but if both methods have the same parameters,  flake8 and pylint are not able to distinguish which of the methods we need at some point of the code. But OK, I should have to add the types.